### PR TITLE
Add from_model_name in Engine

### PR DESCRIPTION
**Pull request by harimkang**
### Summary

Fix CVS-135466
>>> It is not easy or possible to launch the model training using Python API with the given 1) data_root, 2) model_name, 3) task_type, 4) configuration overridings

So for now, Add `Engine.from_model_name` classmethod.

```python
from otx.engine import Engine

model_name = "efficientnet_b0_light"
data_root = "tests/assets/classification_dataset"
task_type = "MULTI_CLASS_CLS"

overriding = {
    "data.config.train_subset.batch_size": 2,
}

engine = Engine.from_model_name(
    model_name=model_name,
    data_root=data_root,
    task=task_type,
    work_dir="otx-workspace-test-1",
    **overriding,
)

assert engine.datamodule.config.train_subset.batch_size == 2
```

However, we're still having trouble overriding the configuration provided by Auto-configuration without using `from_config` or `from_model_name`, so 'll be thinking about how to improve this.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/engine/engine.py [hunk 1]
```diff
@@ -697,6 +697,52 @@ def from_config(
...
+            kwargs: Arguments that can override the engine's arguments.
+
+        Returns:
+            Engine: An instance of the Engine class.
+
+        Example:
+            >>> engine = Engine.from_model_name(
+            ...     model_name="atss_mobilenetv2",
+            ...     task="DETECTION",
+            ...     data_root=<dataset/path>,
```
Comment (vinnamkim) [inline src/otx/engine/engine.py:727]: Please add example about configuration overriding

File: src/otx/engine/engine.py [hunk 2]
```diff
@@ -697,6 +697,52 @@ def from_config(
...
+            ... )
+        """
+        default_config = DEFAULT_CONFIG_PER_TASK.get(task)
+        model_path = str(default_config).split("/")
+        model_path[-1] = f"{model_name}.yaml"
+        config = Path("/".join(model_path))
+        if not config.exists():
+            msg = f"Model config file not found: {config}"
+            raise FileNotFoundError(msg)
```
Comment (vinnamkim) [inline src/otx/engine/engine.py]: Please add available options such as
```python
msg =  f"Model config file not found: {config}. Available model name: {avaliable_model_names}"
```

File: src/otx/engine/engine.py (hunk 1 above)
Comment (harimkang) [reply src/otx/engine/engine.py:727]: Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/3129/commits/8b638c1bc6e46ab989250591dec2b6643978311c

File: src/otx/engine/engine.py (hunk 2 above)
Comment (harimkang) [reply src/otx/engine/engine.py]: Thanks! Done - https://github.com/openvinotoolkit/training_extensions/pull/3129/commits/8b638c1bc6e46ab989250591dec2b6643978311c

**Review by vinnamkim — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Pull request merged**